x/ccv/consumer/keeper/genesis.go

```diff
@@ -37,39 +42,51 @@ func (k Keeper) InitGenesis(ctx sdk.Context, state *consumertypes.GenesisState)
 	}
 
 	// initialValSet is checked in NewChain case by ValidateGenesis
+	// start a new chain
 	if state.NewChain {
-		// Create the provider client in InitGenesis for new consumer chain. CCV Handshake must be established with this client id.
+		// create the provider client in InitGenesis for new consumer chain. CCV Handshake must be established with this client id.
 		clientID, err := k.clientKeeper.CreateClient(ctx, state.ProviderClientState, state.ProviderConsensusState)
 		if err != nil {
 			panic(err)
 		}
 
-		// Set default value for valset update ID
-		k.SetHeightValsetUpdateID(ctx, uint64(ctx.BlockHeight()), uint64(0))
 		// set provider client id.
 		k.SetProviderClientID(ctx, clientID)
-	} else {
-		// verify that latest consensus state on provider client matches the initial validator set of restarted chain
-		// thus, IBC genesis MUST run before CCV consumer genesis
-		consState, ok := k.clientKeeper.GetLatestClientConsensusState(ctx, state.ProviderClientId)
-		if !ok {
-			panic("consensus state for provider client not found. MUST run IBC genesis before CCV consumer genesis")
-		}
-		tmConsState, ok := consState.(*ibctmtypes.ConsensusState)
-		if !ok {
-			panic(fmt.Sprintf("consensus state has wrong type. expected: %T, got: %T", &ibctmtypes.ConsensusState{}, consState))
-		}
 
```

Comment:
Why the switch? Were they not working in the other order?

Reply:
Wdym?